packages/commcare/src/Adaptor.js

```diff
@@ -77,23 +78,62 @@ export function get(path, params = {}, callback = s => s) {
       params
     );
 
+    let { offset = 0, limit = 1000 } = resolvedParams;
+
+    let nextState = state;
+    let result;
+    let allowPagination = !isNaN(resolvedParams.offset);
+
     try {
-      const response = await request(
-        state.configuration,
-        `/a/${domain}/api/v0.5/${resolvedPath}`,
-        {
-          method: 'GET',
-          params: resolvedParams,
-          contentType: 'application/json',
+      let requestParams = {
+        ...resolvedParams,
+      };
+
+      do {
+        const response = await request(
```

Comment:
High level feedback: this do/while loop is kinda complicated. But if it's not too bad if you sort of know what it's doing (get data, check to see if we should auto-page it, if so go again and merge results, otherwise just return the data).  So I'd love to see one or two lines of comment which broadly explain the strategy to the next developer coming into the code.